test/onhost-e2e/ansible/proxy.yaml

```diff
@@ -0,0 +1,171 @@
+        volumes:
+          - /tmp/mitmproxy:/home/mitmproxy/.mitmproxy
+        ports:
+          - "8080:8080"
+          - "8081:8081"
+
+    - name: Wait for mitmproxy CA certificate to be generated
+      ansible.builtin.wait_for:
+        path: /tmp/mitmproxy/mitmproxy-ca.pem
+        timeout: 30
+
+    - name: Copy mitmproxy CA certificate to system trust store
+      ansible.builtin.copy:
+        src: /tmp/mitmproxy/mitmproxy-ca.pem
+        dest: /usr/local/share/ca-certificates/mitmproxy-ca.crt
+        remote_src: yes
+
+    - name: Update CA certificates
+      ansible.builtin.command:
+        cmd: update-ca-certificates
+
+    - name: Install UFW
+      ansible.builtin.package:
+        name: ufw
+        state: present
+
+    - name: Backup current UFW rules
+      shell: |
+        ufw status > /tmp/ufw_status.txt
+        cp /etc/ufw/user.rules /tmp/user.rules.backup
+        cp /etc/ufw/user6.rules /tmp/user6.rules.backup
+
+    - name: Set firewall rules
+      block:
+        - name: Deny all outgoing traffic by default
+          community.general.ufw:
+            policy: deny
+            direction: outgoing
+
+        - name: Allow proxy to reach out from docker
+          community.general.ufw:
+            rule: allow
+            direction: out
+            interface: docker0
+
+        - name: Enable UFW
+          community.general.ufw:
+            state: enabled
+
+        - name: Verify UFW rules with bash script
+          shell: |
+            # Test that direct traffic is blocked
+            curl -s --max-time 5 https://www.newrelic.com
+            direct=$?
+            if [ $direct -eq 0 ]; then
+              echo "ERROR: Direct traffic is not blocked"
+              exit 1
+            fi
+            
+            # Test that proxied traffic works
+            curl -s --max-time 5 --proxy http://localhost:8080 https://www.newrelic.com
+            proxy=$?
+            if [ $proxy -ne 0 ]; then
+              echo "ERROR: Traffic blocked with proxy or proxy not working"
+              exit 1
+            fi
+
+        - name: Install Agent Control
+          environment:
+            HTTPS_PROXY: "http://localhost:8080"
+            HTTP_PROXY: "http://localhost:8080"
+          block:
+            - name: Create AC etc dir
+              file:
+                path: "/etc/newrelic-agent-control/"
+                state: directory
+
+            - name: Import the L2 private key from env-var
+              copy:
+                dest: "/etc/newrelic-agent-control/priv.key"
+                # Read private key from an env var directly because of issues passing multiline --extra-vars to ansible-playbook.
+                content: "{{ lookup('ansible.builtin.env', 'NR_SYSTEM_IDENTITY_PRIVATE_KEY') }}"
+                
+            - name: Install Agent Control
+              include_role:
+                name: install_ac_recipe
+              vars:
+                ac_proxy_url: "http://localhost:8080"
```

Comment:
In `debian.yaml` this variable is passed to `HTTPS_PROXY`, shouldn't we use `http://localhost:8080` then?